addons/mail/controllers/google_translate.py

```diff
@@ -0,0 +1,48 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route, Controller
+
+import babel
+import requests
+
+
+class GoogleTranslateController(Controller):
+    @route("/mail/message/translate", type="json", auth="user")
+    def translate(self, message_id):
+        message = request.env["mail.message"].browse(message_id)
+        message._validate_access_for_current_persona("read")
+        MessageTranslation = request.env["mail.message.translation"].sudo()
+        user_lang = request.env.user.lang.split("_")[0]
+        translation = MessageTranslation.search([("message_id", "=", message_id), ("target_lang", "=", user_lang)])
```

Comment:
We currently don't seem to handle the case of a message update?
I would suggest to drop all the translations of a message when it gets updated.

Reply:
Hopefully this was the right approach: https://github.com/odoo/odoo/pull/126555/files#diff-a90153cd0202b1feb0d5ef9cf041bf662256cb7ce1434b610709a7557db3c9ceR4137-R4138

Note: Split in two lines for the formatter